### DISCUSSION TITLE
Using faster-whisper to translate a text file to text

### DISCUSSION THREAD

**Discussion by kasperkotisaari** [SYSTRAN/faster-whisper#953] [Q&A]
Hi!
Does anybody know if anyone somewhere has succesfully implemented faster-whisper-xxl to do text-to-text translation conversion from source language to English? 

In my use case, I usually want to first use the model to transcribe the source language audio/speech to text, and then manually correct the resulting text. After this, I would like to translate the corrected text to English. I have noticed faster-whisper is comparably good at translation, and I would like to use the model's capabilities for this task too, but I have been unable to use its translation powers without inputing audio to the model. I believe it could be possible to twist it in order to be able to use text as the input for the translation task. Has anybody done this, or is it not possible? Feel free to tell me also if there is some other reason why this would not be the most efficient way to approach this task. 

Thank you very much!

**Comment by Jiltseb** [comment]
Whisper is a Speech to Text model and it can not be used for text to text conversion. You may use any open source translator library instead.

**Reply by kasperkotisaari** [reply]
Thank you for your answer!
However, what I'm thinking is that at some point of the model's functioning, it has predicted words from speech audio in, let's say for example, Finnish language. Now, the model's translation part steps in and tries to predict English words corresponding to the Finnish words found from the audio - isn't this still correct? 
Now, shouldn't I be able to trick the model to believe the words to be translated are a result of the audio processing, without actually having to give the model any audio? This is what I believe is possible, and would like to know if someone has tried to do it :)
